### ISSUE TITLE
Allow flexible kafka URL for publishing/subscribing to multiple topics

### ISSUE THREAD

**Issue by bfc5288** [scimma/hop-client#70]
As a developer/user, I would like to use `hop publish/subscribe` for multiple topics by specifying either of two URLs, e.g.,

1) `kafka://dev.hop.scimma.org:9092/topic1,topic2,topic3`
or 
2) `kafka://dev.hop.scimma.org/topic1,topic2,topic3`

Currently, `publish/subscribe` work for both (1) and (2) when using only one topic, but multiple topics are supported only with (1).

DoD:

- [ ] Enable publishing to multiple topics when not specifying port number in the kafka URL
- [ ] Enable subscribing to multiple topics when not specifying port number in the kafka URL

**Comment by spenczar** [comment]
This kind of sounds like a bug in `adc-streaming`, at least for subscribe. Seems worth fixing.

**Comment by mjuric** [comment]
Definitely a bug, not a feature.

**Comment by myNameIsPatrick** [comment]
@bfc5288, do you have a good way of reproducing this? I don't have a good mechanism for testing this on a URL that doesn't have a port specified.

**Comment by aepace** [comment]
@bfc5288 I'm seeing some unexpected behavior that might be related to this. 

Here's the scenario: subscribe and listen to two topics, for example:

```
with stream.open('kafka://kafka.scimma.org/topic1,topic2', 'r') as s:
    for payload in s:
        print(payload)
```

Separately, send a message to topic1 and topic2. I'm only seeing the message for topic2. Reverse the order:

```
with stream.open('kafka://kafka.scimma.org/topic2,topic1', 'r') as s:
    for payload in s:
        print(payload)
```

Send a message to topic1 and topic2 (alternatively topic2 then topic1), and only the message for topic1 is received.  

I'm seeing this whether or not the port is explicitly stated, too. 

Reproducible with the API and the CLI, `hop-client==0.4.0` from `pip`.

**Comment by myNameIsPatrick** [comment]
I can reproduce this issue and have identified the problem. The problem has to do with topic subscription, where making multiple calls to subscribe replaces the previous subscription. The result is that only the latest topic to be subscribed is the only one that the client listens to.

This needs changes both here and in `adc-streaming`. I'll be opening merge requests shortly with fixes.

**Comment by myNameIsPatrick** [comment]
@aepace, this should be addressed by #149. Let me know if this resolves the problem you've been seeing.
